### DISCUSSION TITLE
Increase default request timeout for Adonis 5

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#3977] [Help] [answered]
Hey 👋 Is there a way to increase the default timeout (Error code 504) with Adonis 5?

It's probably a setting to change somewhere, but I can't find it. I have a query that takes between 2 and 3 minutes to complete before responding. In Adonis 4, it was something close to this.

`Server.getInstance().timeout = 0`

https://github.com/adonisjs/core/issues/877

**Comment by thetutlage** [comment, marked as answer]
You can do the same thing with v5. Write the following code within the `AppProvider` of your app.

```ts
  public async ready() {
    if (this.app.environment === 'web') {
      this.app.container.use('Adonis/Core/Server').instance!.timeout = 0
    }
  }

```

**Reply by LeCoupa** [reply]
Thank you @thetutlage

**Reply by navinrangar** [reply]
> You can do the same thing with v5. Write the following code within the `AppProvider` of your app.
> 
> ```ts
>   public async ready() {
>     if (this.app.environment === 'web') {
>       this.app.container.use('Adonis/Core/Server').instance!.timeout = 0
>     }
>   }
> ```

in case it's an API server, should I write `if (this.app.environment === 'api') `  or the `if (this.app.environment === 'web') ` is fine.
